src/allmydata/web/root.py

```diff
@@ -182,9 +183,44 @@ def data_import_path(self, ctx, data):
     def render_my_nodeid(self, ctx, data):
         tubid_s = "TubID: "+self.client.get_long_tubid()
         return T.td(title=tubid_s)[self.client.get_long_nodeid()]
+
     def data_my_nickname(self, ctx, data):
         return self.client.nickname
 
+    def renderHTTP(self, ctx):
+        req = inevow.IRequest(ctx)
+        t = get_arg(req, "t")
+        if t == "json":
+            return self.json_welcome(ctx)
+        return rend.Page.renderHTTP(self, ctx)
+
+    def json_welcome(self, ctx):
+        inevow.IRequest(ctx).setHeader("content-type", "text/plain")
```

Comment:
Probably just me being foolish. I liked being able to see the JSON in a web browser, and my experiments with `application/json` indicated that it wouldn't be shown if we used that. If `text/json` is displayed (rather than being forced to be downloaded), I'd be happy with it.